### PR TITLE
Default implementation for GetSubQueryMarker

### LINKED ISSUE #3919: ProjectTo throw NullReferenceException
Hi,

I tried to upgrade AutoMapper from version 10 to 11 and after that my tests failed with NullReferenceException from AutoMapper.

I identified that the issue is caused by the global mapping that I use to convert Null string to Empty string. [This method](https://stackoverflow.com/a/23086166) did work up to version 10.
Standard mapping is not impacted but some ProjectTo now fail.

### Source/destination types
See LinqPad reproduction in attachment.

### Mapping configuration
See LinqPad reproduction in attachment.

### Version: x.y.z
AutoMapper version: either 11.0.0 or 11.0.1 (10.x version are not impacted)
EFcore version : either 5.x or 6.x (not Efcore dependant)

### Expected behavior
The ProjectTo method work without issue.

### Actual behavior
The ProjectTo method throw an NullReferenceException.

### Steps to reproduce
LinqPad reproduction in attachment, drop it to LinqPad to execute.

[ProjectToIssue.txt](https://github.com/AutoMapper/AutoMapper/files/8163094/ProjectToIssue.txt)

### PR DESCRIPTION
**Pull request by lbargaoanu**
Fixes #3919.

### CODE REVIEW

**Review by jbogard — APPROVED** [review]

File: src/IntegrationTests/ProjectionAdvanced.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using AutoMapper.UnitTests;
+using Microsoft.EntityFrameworkCore;
+using Shouldly;
+using System.Linq;
+using Xunit;
+namespace AutoMapper.IntegrationTests;
+public class ProjectionAdvanced : IntegrationTest<ProjectionAdvanced.Initializer>
+{
+    protected override MapperConfiguration CreateConfiguration() => new(c => c.CreateProjection<Entity, Dto>().Advanced().ForAllMembers(o=>o.Ignore()));
```
Comment (lbargaoanu) [inline src/IntegrationTests/ProjectionAdvanced.cs:9]: I'm not sure what extra members should we _really_ have for projections. But this is a decent workaround I believe. From #3929.

**Comment by jbogard** [comment]
This ready to merge? Go ahead whenever

**Comment by lbargaoanu** [comment]
It's ready, but I was thinking we would release another patch for 11 :) But #3895 is a binary breaking change. Granted, a small one, but breaking nevertheless :)

File: src/UnitTests/ExtensionMethods.cs [hunk 2]
```diff
@@ -7,6 +7,47 @@
...
+            public int Count { get; set; }
+        }
+        protected override MapperConfiguration CreateConfiguration() => new(c =>
+        {
+            c.IncludeSourceExtensionMethods(typeof(GenericExtensions));
+            c.CreateMap<Source, Destination>();
+        });
+        [Fact]
+        public void It_should_fail_validation() => new Action(AssertConfigurationIsValid).ShouldThrow<AutoMapperConfigurationException>()
+            .Errors[0].UnmappedPropertyNames[0].ShouldBe(nameof(Destination.Count));
```
Comment (lbargaoanu) [inline src/UnitTests/ExtensionMethods.cs:27]: Letting the exception escape would result in this test failing on `Seal` with the initial error. I wouldn't mind that :) Certainly easier to understand what's going on.

Comment (jbogard) [reply src/UnitTests/ExtensionMethods.cs:27]: Ha yeah, definitely

**Comment by lbargaoanu** [comment]
@jbogard I think we can merge this (I cannot do it myself even when approved).

**Pull request merged**